Jint.Tests/Runtime/ProxyTests.cs

```diff
@@ -183,4 +186,218 @@ public void ConstructHandlerInvariant()
+    {
+        _engine.SetValue("testClass", TestClass.Instance);
+        var result = _engine.Evaluate("""
+            const handler = {
+              get(target, property, receiver) {
+                return Reflect.get(target, property, receiver);
+              }
+            };
+            const p = new Proxy(testClass, handler);
+            return p.StringValue;
+        """);
+        Assert.Equal(TestClass.Instance.StringValue, result.AsString());
+    }
+
+    [Fact]
+    public void ProxyClrPropertyPrimitiveInt()
+    {
+        _engine.SetValue("testClass", TestClass.Instance);
+        var result = _engine.Evaluate("""
+            const handler = {
+              get(target, property, receiver) {
+                return Reflect.get(target, property, receiver);
+              }
+            };
+            const p = new Proxy(testClass, handler);
+            return p.IntValue;
+        """);
+        Assert.Equal(TestClass.Instance.IntValue, result.AsInteger());
+    }
+
+    [Fact]
+    public void ProxyClrPropertyObjectWrapper()
+    {
+        _engine.SetValue("testClass", TestClass.Instance);
+        var result = _engine.Evaluate("""
+            const handler = {
+              get(target, property, receiver) {
+                return Reflect.get(target, property, receiver);
+              }
+            };
+            const p = new Proxy(testClass, handler);
+            return p.ObjectWrapper;
+        """);
+    }
+
+    private static ErrorPrototype TypeErrorPrototype(Engine engine)
+        => engine.Realm.Intrinsics.TypeError.PrototypeObject;
+
+    private static void AssertJsTypeError(Engine engine, JavaScriptException ex, string msg)
+    {
+        Assert.Same(TypeErrorPrototype(engine), ex.Error.AsObject().Prototype);
+        Assert.Equal(msg, ex.Message);
+    }
+
+    // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy/Proxy/get#invariants
+    // The value reported for a property must be the same as
+    // the value ofthe corresponding target object property,
+    // if the target object property is
+    // a non-writable, non-configurable own data property.
+    [Fact]
+    public void ProxyHandlerGetInvariantsDataPropertyReturnsDifferentValue()
+    {
+        _engine.Execute("""
+            let o = Object.defineProperty({}, 'value', {
+              writable: false,
+              configurable: false,
+              value: 42,
+            });
+            const handler = {
+              get(target, property, receiver) {
+                return 32;
+              }
+            };
+            let p = new Proxy(o, handler);
+        """);
+        var ex = Assert.Throws<JavaScriptException>(() => _engine.Evaluate("p.value"));
+        AssertJsTypeError(_engine, ex, "'get' on proxy: property 'value' is a read-only and non-configurable data property on the proxy target but the proxy did not return its actual value (expected '42' but got '32')");
```

Comment:
Should we comment all `AssertJsTypeError`?  
As it depends internal implementations and unstable error messages.  
Just comment out these for develop usage.

Reply:
I think you did great work replicating the V8 error messages and they can stay, we can always tweak later if comes a maintenance burden 👍🏻